src/Raven.Client/Documents/Changes/DatabaseChanges.cs

```diff
@@ -695,47 +722,53 @@ private async Task ProcessChanges()
             }
         }
 
-        private void NotifySubscribers(string type, BlittableJsonReaderObject value, List<DatabaseConnectionState> states)
+        private void NotifySubscribers(string type, BlittableJsonReaderObject value)
         {
             switch (type)
             {
+                case nameof(AggressiveCacheUpdate):
+                    foreach (var state in _counters)
```

Comment:
shouldn't we call 'ForceEnumerateInThreadSafeManner' ? I think it was important here

Reply:
`ForceEnumerateInThreadSafeManner` does the exact same thing. That is why we are iterating using `foreach` here.
Anything else will force you to take locks and materialize the values.